include/odp/api/spec/timer.h

```diff
@@ -357,6 +357,54 @@ uint64_t odp_timer_ns_to_tick(odp_timer_pool_t timer_pool, uint64_t ns);
  */
 uint64_t odp_timer_current_tick(odp_timer_pool_t timer_pool);
 
+/**
+ * Timer tick information
+ */
+typedef struct odp_timer_tick_info_t {
+	/**
+	 * Timer tick frequency in hertz
+	 *
+	 * Timer tick frequency expressed as a fractional number. The integer part contains
```

Comment:
I am still not clear why fractional description should be used and not floating-point. The number should show the exact frequency (or period) of clock, which feeds the timer hardware, and floating-point can express that. If the role of numerator and denominator is to show the relationship between input and timer clocks, then two numbers may not be enough, because, in general, such relations may be more complex.

Reply:
This is not frequency of the clock signal, but frequency of the timer tick (in the API). For example, if tick_info tells that timer tick is 100MHz, application can request timeout every 100 ticks and get exact 1MHz timeout frequency. 

Fractional presentation is exact on the relation between nsec and hz, e.g. when tick period is 1200nsec the  frequency is 833 333 1/3 Hz. It's up to implementation if it rounds that freq to 6 digit granularity 833 333 333333/1000000 (833333.333333), or if it outputs the exact ratio (1/3). With float in API, implementation could not output the exact ratio.

Implementation can express relation between tick (in API) and source clock signal accurately, as long as it a series of multipliers and dividers: (2 * 128) / 3 = 85 1/3